File: plugins/slopocop-code/skills/code-slop/scripts/tests/fixtures/slop.go
package fixtures

import "strconv"

// =====================================================
// Number summation helpers
// =====================================================

// AbstractNumberSummationStrategyProvider defines a summer
type AbstractNumberSummationStrategyProvider interface {
	Sum(data []string) (int, error)
}

// DefaultSummer implements the summer interface
type DefaultSummer struct{}

// Sum sums the data and returns the result
func (d DefaultSummer) Sum(data []string) (int, error) {
	// Initialize result to zero
	result := 0
	// Loop over the data slice
	for _, item := range data {
		// Convert the item to an integer
		temp, err := strconv.Atoi(item)
		// Check if there was an error
		if err != nil {
			// Return the error
			return 0, err
		}
		// Add temp to result
		result = result + temp
	}
	// 🎯 Return the result
	return result, nil
}

// -----------------------------------------------------
func SumAgainButWithADifferentFunctionName(data []string) (int, error) {
	result := 0
	for _, item := range data {
		temp, err := strconv.Atoi(item)
		if err != nil {
			return 0, err
		}
		result = result + temp
	}
	return result, nil
}

File: plugins/slopocop-code/skills/code-slop/scripts/tests/fixtures/slop.ts
// ==========================================================
// User data processing service
// ==========================================================

// This interface defines the shape of a processor
interface AbstractDataProcessorInterfaceDefinition {
  process(data: number[]): number;
}

// This class implements the processor interface
class DataProcessorImplementation implements AbstractDataProcessorInterfaceDefinition {
  // This method processes the data and returns the result
  process(data: number[]): number {
    // Initialize result to zero
    let result = 0;
    // Loop over the data array
    for (const item of data) {
      // Add item to result
      const temp = item;
      result = result + temp;
    }
    // 🚀 Return the final result
    return result;
  }
}

// ----------------------------------------------------------
function sumThatIsAlsoUsedSomewhereElseEntirely(data: number[]): number {
  let result = 0;
  for (const item of data) {
    const temp = item;
    result = result + temp;
  }
  return result;
}

export { DataProcessorImplementation, sumThatIsAlsoUsedSomewhereElseEntirely };
